drivers/SmartThings/deepsmart/src/commands.lua

```diff
@@ -93,6 +93,23 @@ function command_handler.set_thermostat_mode(driver, device, command)
   log.error('no response from device')
 end
 
+----------------
+-- heater mode command
+----------------
+function command_handler.set_thermostat_mode(driver, device, command)
+  -- send command
+  local success = wisers.control(device, command, {config.HEATER.ONOFF})
+  -- Check if success
+  if success then
+    if command.arg.mode == 'off' then
+      return device:emit_event(caps.switch.switch.off())
+    end
+    return device:emit_event(caps.switch.switch.on())
```

Comment:
Why are switch events emitted here? The app will timeout on set thermostat mode commands if no thermostat mode event is emitted, which is a bad user experience. Perhaps either ThermostatMode should not be a part of the device profile? https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference/#thermostatMode

Reply:
> Perhaps either ThermostatMode should not be a part of the device profile? https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference/#thermostatMode

Hi @cjswedes 

profiles/Heater.yml is used for Underfloor Heating Thermostats devices.
So we set category of profile to *Thermostat*, we also noticed [Required-capabilities](https://developer.smartthings.com/docs/certification/required-capabilities) url.

There are 4 required-capabilities of *Thermostat*:

1. temperatureMeasurement
2. thermostatMode
3. thermostatCoolingSetpoint or thermostatHeatingSetpoint or both
4. thermostatOperatingState 

So I think we shouldn't remove "ThermostatMode" from profile.

@refuseRed 

The overall structure is that,    

ST cloud <----Send Capability----> ST Hub <----Call REST API----> DeepSmart bridge -- Bridged devices.

In most cases we should convert the capability to DeepSmart restAPI in capability handler.

"ThermostatMode" has supported *off* and *heat* commands already, 
I think we could replace "switch" capability with above 2 commands.

Thanks.